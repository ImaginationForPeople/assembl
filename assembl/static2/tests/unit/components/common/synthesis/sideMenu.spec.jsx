import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import SideMenu from '../../../../../js/app/components/synthesis/sideMenu';
import '../../../../helpers/setupTranslations';

describe('SideMenu component', () => {
  it('should render a SideMenu component', () => {
    const props = {
      rootIdeas: [
        {
          ancestors: [],
          id: '2222222',
          img: {
            externalUrl: 'http://foo.com/bar'
          },
          numContributors: 10,
          numPosts: 100,
          phaseIdentifier: 'phaseId',
          posts: {
            edges: []
          },
          synthesisTitle: 'Intersting idea',
          title: 'Cool story'
        }
      ],
      descendants: [
        {
          ancestors: [],
          id: '33333',
          img: {
            externalUrl: 'http://foo.com/bar'
          },
          numContributors: 5,
          numPosts: 60,
          phaseIdentifier: 'phaseId',
          posts: {
            edges: []
          },
          synthesisTitle: 'Very intersting idea',
          title: 'Super cool story'
        }
      ],
      synthesisPostId: '5667'
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<SideMenu {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});