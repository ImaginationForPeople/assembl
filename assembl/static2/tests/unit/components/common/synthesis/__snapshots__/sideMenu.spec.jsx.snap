// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu component should render a SideMenu component 1`] = `
<div
  className="synthesis-side-menu"
>
  <Translate
    className="dark-title-4"
    value="synthesis.tableOfContents"
  />
  <div
    className="title-hyphen block"
  >
    Â 
  </div>
  <SideMenuTree
    ideaOnScroll={null}
    index={1}
    indexGenerator={[Function]}
    parents={Array []}
    rootIdea={
      Object {
        "ancestors": Array [],
        "id": "2222222",
        "img": Object {
          "externalUrl": "http://foo.com/bar",
        },
        "numContributors": 10,
        "numPosts": 100,
        "phaseIdentifier": "phaseId",
        "posts": Object {
          "edges": Array [],
        },
        "synthesisTitle": "Intersting idea",
        "title": "Cool story",
      }
    }
    slug={null}
    subIdeas={Array []}
    synthesisPostId="5667"
  />
</div>
`;
